crates/edr_evm/src/blockchain/local.rs

```diff
@@ -64,75 +87,56 @@ impl LocalBlockchain {
     #[cfg_attr(feature = "tracing", tracing::instrument(skip_all))]
     #[allow(clippy::too_many_arguments)]
     pub fn new(
-        genesis_diff: StateDiff,
+        mut genesis_diff: StateDiff,
         chain_id: u64,
         spec_id: SpecId,
-        gas_limit: u64,
-        timestamp: Option<u64>,
-        prevrandao: Option<B256>,
-        base_fee: Option<U256>,
-        blob_gas: Option<BlobGas>,
-        parent_beacon_block_root: Option<B256>,
+        options: GenesisBlockOptions,
     ) -> Result<Self, CreationError> {
         const EXTRA_DATA: &[u8] = b"\x12\x34";
 
+        if spec_id >= SpecId::CANCUN {
```

Comment:
Add the beacon roots contract in the local blockchain. Not needed for the forked one, as it should be present remotely.

Reply:
Can it happen that a fork is running with a pre-Cancun block number, but with a Cancun block spec? This could mean that the beacon roots contract is not available in the forked blockchain. cc @fvictorio

Reply:
I think you are right. I will add tests for this.

Reply:
Uhm, I [added tests](https://github.com/NomicFoundation/hardhat/pull/4889/commits/541717193deb3f1700fcef568dca07f0631870a9) for that scenario and they are working fine :man_shrugging:

(Those tests are skipped right now because they assume that the hardfork is enabled in mainnet, but they can be run locally by setting a sepolia URL and using the commented-out sepolia blocks.)

Anyway, I'm ok with merging this as is.